src/platform/plugins/shared/saved_objects_management/public/management_section/objects_table/components/table.tsx

```diff
@@ -369,26 +373,64 @@ export class Table extends PureComponent<TableProps, TableState> {
       );
     }
 
-    const button = (
-      <EuiButton
-        iconType="arrowDown"
-        iconSide="right"
-        onClick={this.toggleExportPopoverVisibility}
-        isDisabled={selectedSavedObjects.length === 0}
-      >
-        <FormattedMessage
-          id="savedObjectsManagement.objectsTable.table.exportPopoverButtonLabel"
-          defaultMessage="Export"
-        />
-      </EuiButton>
-    );
-
     const activeActionContents = this.state.activeAction?.render() ?? null;
     const exceededResultCount = totalItemCount > MAX_PAGINATED_ITEM;
 
+    const hasMlObjects = selectedSavedObjects.some(({ type }) => ML_SAVED_OBJECT_TYPES.has(type));
+
     const anySelected = selectedSavedObjects.length > 0;
     const allHidden =
       anySelected && selectedSavedObjects.every(({ meta: { hiddenType } }) => hiddenType);
+
+    const deleteTooltip = () => {
+      if (hasMlObjects) {
+        return (
+          <FormattedMessage
+            id="savedObjectsManagement.objectsTable.table.hasMlObjects.deleteDisabledTooltip"
+            defaultMessage="Navigate to the Machine Learning management page to delete machine learning objects."
+          />
+        );
+      }
+      if (allHidden) {
+        return (
+          <FormattedMessage
+            id="savedObjectsManagement.objectsTable.table.deleteDisabledTooltip"
+            defaultMessage="Selected objects can’t be deleted because they are hidden objects."
+          />
+        );
+      }
+    };
+    const disabledExportTooltip = () => {
+      if (hasMlObjects) {
+        return (
+          <FormattedMessage
+            id="savedObjectsManagement.objectsTable.table.hasMlObjects.exportDisabledTooltip"
+            defaultMessage="Navigate to the Machine Learning management page to export machine learning objects."
+          />
+        );
+      }
+    };
+    const button = (
+      <EuiToolTip
+        data-test-subj="exportSOToolTip"
+        key="exportSOToolTip"
+        content={disabledExportTooltip()}
+      >
+        <EuiButton
+          iconType="arrowDown"
+          iconSide="right"
+          onClick={this.toggleExportPopoverVisibility}
+          isDisabled={selectedSavedObjects.length === 0}
+          disabled={hasMlObjects}
```

Comment:
Is it possible to disable the 'Export all' button in the top nav if only ML objects are listed? The current workflow leads to the following error toast:

<img width="1702" alt="Screenshot 2025-01-07 at 17 39 52" src="https://github.com/user-attachments/assets/3802fe30-44f2-4b69-bb81-fab90643f445" />